Sources/MapboxNavigation/NavigationViewControllerDelegate.swift

```diff
@@ -113,6 +113,37 @@ public protocol NavigationViewControllerDelegate: VisualInstructionDelegate {
      */
     func navigationViewController(_ navigationViewController: NavigationViewController, didRerouteAlong route: Route)
     
+    /**
+     Called when navigation view controller has detected user taking an alternative route.
+     
+     This method is called before updating main route.
+     
+     - parameter navigationViewController: The navigation view controller that has detected turning to the alternative.
+     - parameter route: The alternative route which will be taken as new main.
+     - parameter location: The user’s current location.
+     */
+    func navigationViewController(_ navigationViewController: NavigationViewController, willTakeAlternativeRoute route: Route, at location: CLLocation?)
+    
+    /**
+     Called when navigation view controller has finished switching to an alternative route
+     
+     This method is called after `navigationViewController(_:willTakeAlternativeRoute:)`
+     
+     - parameter navigationViewController: The navigation view controller that switched to the alternative.
+     - parameter location: The user’s current location.
+     */
+    func navigationViewController(_ navigationViewController: NavigationViewController, didTakeAlternativeRouteAt location: CLLocation?)
+    
+    /**
+     Called when navigation view controller has failed to take an alternative route.
```

Comment:
Clarify why this method would get called. Presumably it doesn’t get called just because the _end user_ fails to take the suggested alternative route?

Reply:
Appended to the code doc